### PR TITLE
Specs for CLI class, bunch more refactors and adding daemon functionality

### PR DESCRIPTION
**Pull request by murny**
#18 

* Refactor CLI stuff and add some spec tests for it
* Extract default options into a config class
* Setup logger for PushmiPullyu
* Add daemon support using Daemons gem

### CODE REVIEW

File: exe/pushmi_pullyu [hunk 1]
```diff
@@ -2,4 +2,13 @@
...
 require 'pushmi_pullyu'
 
-PushmiPullyu::CLI.new(ARGV).run!
+begin
+  cli = PushmiPullyu::CLI.instance
+  cli.parse
+  cli.run
+rescue => e
+  raise e if $DEBUG
+  STDERR.puts e.message
```
Comment (mbarnett) [inline exe/pushmi_pullyu]: Are we just going to log to STDERR, or are we deferring logging to a later ticket?

File: lib/pushmi_pullyu/logging.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'logger'
+
+# PushmiPullyu::Logging is a standard Ruby logger wrapper
+module PushmiPullyu::Logging
```
Comment (mbarnett) [inline lib/pushmi_pullyu/logging.rb]: It occurred to me, if this is going to be a long-running daemon, we need to trap SIGHUP and swap out the logs to play nicely with log-rotation automation

File: exe/pushmi_pullyu (hunk 1 above)
Comment (mbarnett) [inline exe/pushmi_pullyu]: Nevermind, saw the logging class after. Are we intentionally bypassing it here?

Comment (murny) [inline exe/pushmi_pullyu]: Yeah I'll get rid of this logic rescue logic. Not really needed.

File: lib/pushmi_pullyu/cli.rb [hunk 3]
```diff
@@ -18,26 +18,47 @@ def initialize
...
+      exit(0)
     end
   end
 
   private
 
+  def setup_signal_traps
+    Signal.trap('INT') { raise Interrupt }
+    Signal.trap('TERM') { raise Interrupt }
+    Signal.trap('HUP') do
```
Comment (mbarnett) [inline lib/pushmi_pullyu/cli.rb]: 👍

Comment (murny) [inline lib/pushmi_pullyu/cli.rb]: I'll merge as is...but still figuring out how to reopen/rotate logs on SIGHUPs. So ill add another PR for this. As this one is getting pretty beefy as is

**Pull request merged**